gnd/src/main/java/com/google/android/gnd/ui/home/featureselector/FeatureSelectorFragment.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.ui.home.featureselector;
+
+import static com.google.android.gnd.rx.RxAutoDispose.autoDisposable;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java8.util.stream.StreamSupport.stream;
+
+import android.app.AlertDialog;
+import android.app.AlertDialog.Builder;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import com.google.android.gnd.R;
+import com.google.android.gnd.databinding.FeatureSelectorDialogBinding;
+import com.google.android.gnd.model.feature.Feature;
+import com.google.android.gnd.ui.common.AbstractDialogFragment;
+import com.google.android.gnd.ui.common.EphemeralPopups;
+import com.google.common.collect.ImmutableList;
+import dagger.hilt.android.AndroidEntryPoint;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import org.jetbrains.annotations.NotNull;
+
+@AndroidEntryPoint
+public class FeatureSelectorFragment extends AbstractDialogFragment {
+  @Inject EphemeralPopups popups;
+
+  private FeatureSelectorViewModel viewModel;
+
+  @SuppressWarnings("NullAway")
+  private FeatureSelectorDialogBinding binding;
+
+  @Nullable private ArrayAdapter listAdapter;
+
+  public FeatureSelectorFragment(FeatureSelectorViewModel featureSelectorViewModel) {
+    this.viewModel = featureSelectorViewModel;
+  }
+
+  @Override
+  public @NotNull Dialog onCreateDialog(Bundle savedInstanceState) {
```

Comment:
`@NotNull` is the default in our codebase so not needed here.

Reply:
Thanks!